scripts/irods/test/test_rule_engine_plugin_framework.py

```diff
@@ -81,6 +85,86 @@ def test_bulk_put_operation__5164(self):
             self.admin.assert_icommand(['irm', '-r', '-f', logical_path])
             shutil.rmtree(local_dir)
 
+
+    @unittest.skipIf(plugin_name != 'irods_rule_engine_plugin-python' or test.settings.RUN_IN_TOPOLOGY, "Skip for Python REP and Topology Testing")
+    def test_remote_native_rule_engine_plugin_instance_6465(self):
+        with append_native_re_to_server_config():
+            with temporary_core_file( IRODS_RULE_LANGUAGE_RULE_ENGINE_PLUGIN_NAME ) as f_ir:
+                f_ir.add_rule("""
+                mymain() {
+                    writeLine('serverLog', 'Hello from Native Rule Engine Plugin')
+                }
+                """)
+                with tempfile.NamedTemporaryFile(suffix=".r",mode="w") as f:
+                    f.write(textwrap.dedent("""\
+                        main() {
+                            remote("localhost", "<INST_NAME>irods_rule_engine_plugin-irods_rule_language-instance</INST_NAME>") {
+                              mymain
+                            }
+                        }
+                        INPUT null
+                        OUTPUT null
+                        """))
+                    f.flush()
+                    log_offset = lib.get_file_size_by_path(paths.server_log_path())
+                    self.admin.assert_icommand(['irule', '-r', 'irods_rule_engine_plugin-irods_rule_language-instance', '-F', f.name])
+
+                    # Assert the core.re rule ran and had the desired effect.
+                    lib.delayAssert(
+                        lambda: lib.log_message_occurrences_equals_count(
+                            msg="Hello from Native",
+                            start_index=log_offset))
+
+                    # The following sleep seems to be the best way, presently, to assert that the unwelcome warning did not happen.
+                    # (We could use delayAssert to ensure a count of zero, but it obfuscates and has same effect as the sleep.)
+                    time.sleep(2)
+
+                    # Assert we don't see a warning from the Python plugin
+                    self.assertTrue(
+                        lib.log_message_occurrences_equals_count(
+                            msg='Improperly formatted rule text in Python rule engine plugin',
+                            count=0,
+                            start_index=log_offset))
+
+
+    @unittest.skipIf(plugin_name != 'irods_rule_engine_plugin-python' or test.settings.RUN_IN_TOPOLOGY, "Skip for Python REP and Topology Testing")
+    def test_remote_python_rule_engine_plugin_instance_6465(self):
+        with append_native_re_to_server_config():
```

Comment:
is this correct?   append_native?

Reply:
Yes.. We are allowing native and PREP to co-exist in the configuration for purposes of the test.